internal/pkg/component/gitlab/gitlab.go

```diff
@@ -168,9 +194,55 @@ func (c *Component) GetAPIEndpoint() string {
 	return fmt.Sprintf("https://%s/api/v4/", c.Host)
 }
 
+func (c *Component) getProjectPathFromRepoUrl(repoUrl string) (string, error) {
+	url, err := url.Parse(repoUrl)
+	if err != nil {
+		return "", err
+	}
+
+	return strings.TrimPrefix(
+		strings.TrimSuffix(url.Path, ".git"),
+		"/",
+	), nil
+}
+
+func (c *Component) GetBaseUrl(repoUrl string) (string, error) {
+	url, err := url.Parse(repoUrl)
+	if err != nil {
+		return "", FailedToParseUrlError{url: repoUrl, err: err.Error()}
+	}
+
+	if url.Scheme == "" {
+		return "", MissingSchemaError{repoUrl}
+	}
+
+	if url.Host == "" {
+		return "", MissingHostError{repoUrl}
+	}
+
+	// The gitlab client library expects the base url to have a tailing slash
+	return fmt.Sprintf("%s://%s/", url.Scheme, url.Host), nil
+}
+
 func (c *Component) getDefaultBranch() (string, error) {
-	// TODO: call github APIs to determine the default branch
-	return "main", nil
+	token, err := c.GetToken()
+	if err != nil {
+		return "", fmt.Errorf("failed to get GitLab token: %w", err)
+	}
+	baseUrl, err := c.GetBaseUrl(c.Repository)
```

Comment:
This might be misleading, but `c.Repository` is actually the repo path, not the git URL: https://github.com/konflux-ci/mintmaker/blob/fc51f926095d3ba47b343d56a9d7951b4b035bee/internal/pkg/component/base/base.go#L30-L32